index.js

```diff
@@ -1,5 +1,20 @@
 const Multiaddr = require('multiaddr')
+
 const reduceValue = (_, v) => v
+const tcpUri = (str, port, parts, opts) => {
+  // return tcp when explicitly requested
+  if (opts && opts.assumeHttp === false) return `tcp://${str}:${port}`
+  // check if tcp is the last protocol in multiaddr
```

Comment:
Why does it have to be the last protocol?

Reply:
The last protocol is the one that lands in URI. It will become  `foo` in `foo://`. 

If we convert things that are not terminating (last protocol), then lings like `/ip4/1.2.3.4/tcp/3456/ipfs/QmcgpsyWgH8Y8ajJz1Cu72KnS5uo2Aa2LpzU7kinSoooo4`

will get `http://` instead of `tcp://1.2.3.4:3456/ipfs/QmcgpsyWgH8Y8ajJz1Cu72KnS5uo2Aa2LpzU7kinSoooo4`